frontend/src/components/maps/leaflet/Map.tsx

```diff
@@ -165,13 +186,27 @@ const Map: React.FC<MapProps> = ({
   // the same ID could be found on both the parcel collection and building collection
   const generateKey = (p: IProperty) => `${p.propertyTypeId === 0 ? 'parcel' : 'building'}-${p.id}`;
 
-  const renderMarker = (p: IProperty) => (
-    <Marker
-      key={generateKey(p)}
-      position={[p.latitude, p.longitude]}
-      onClick={(e: any) => onMarkerClick?.(p)}
-    />
-  );
+  const renderMarker = (p: IProperty) => {
+    if (p.propertyTypeId === 0) {
+      return (
+        <Marker
+          key={generateKey(p)}
+          position={[p.latitude, p.longitude]}
+          onClick={(e: any) => onMarkerClick?.(p)}
+          icon={greenIcon}
+        />
+      );
+    } else {
+      return (
+        <Marker
+          key={generateKey(p)}
+          position={[p.latitude, p.longitude]}
+          onClick={(e: any) => onMarkerClick?.(p)}
+          icon={blueIcon}
+        />
+      );
+    }
+  };
```

Comment:
I was going to approve as-is as I'd love to see this change merged sooner than later.
One minor observation: I'd prefer to see a simpler logic where we don't duplicate the Marker logic in the `if/then/else`. A simpler approach would lead to less opportunities for future bugs where one branch of the if/then/else is modified but not the other...
Something like this:

```tsx
const icon = p.propertyTypeId === 0 ? greenIcon : blueIcon;
<Marker
    key={generateKey(p)}
    position={[p.latitude, p.longitude]}
    onClick={(e: any) => onMarkerClick?.(p)}
    icon={icon}
/>

```

Reply:
I like this! Will adjust